test/reduce.js

```diff
@@ -426,6 +477,12 @@ function is_statement(node) {
     return node instanceof U.AST_Statement && !(node instanceof U.AST_Function);
 }
 
+function to_sequence(expressions) {
+    if (expressions.length == 0) return new U.AST_Number({value: 0, start: {}});
+    if (expressions.length == 1) return expressions[0];
+    return new U.AST_Sequence({expressions: expressions, start: {}});
```

Comment:
Would this become an issue if elements of `expressions` contains `AST_Sequence`?

`make_sequence()` handles it like this:

https://github.com/mishoo/UglifyJS2/blob/f0a29902acc24217881529449c7f9b8815896297/lib/compress.js#L954-L959
https://github.com/mishoo/UglifyJS2/blob/f0a29902acc24217881529449c7f9b8815896297/lib/compress.js#L1014-L1021

Reply:
I wanted to keep it simple. I ran the PR against dozens of ufuzz induced errors in xor, collapse_vars and reduce_vars and never saw the case you're describing come up. But I'll add the logic you mentioned. It will require retesting.

Reply:
I made these changes locally but I am unable to come up with a test to exercise it.  I suspect what happened before is that the unneeded sequence elements were just dropped by normal `reduce_test` operations, and single-element sequences were already converted to a regular expression. Nevertheless, I'll keep the changes and continue to test with it.

Reply:
You may be correct that it was only done internally to ensure optimal compression.

Thanks for entertaining my worries nonetheless 👍

Reply:
No problem. 

Not much left to optimize in reduce_test. It's good enough.